install.sh

```diff
@@ -1,37 +1,121 @@
 #!/usr/bin/env bash
 
-set -e
+# Source: nvm (project)
+try_profile() {
+  if [ -z "${1-}" ] || [ ! -f "${1}" ]; then
+    return 1
+  fi
+  echo "${1}"
+}
+
+# Source: nvm (project)
+detect_profile() {
+  if [ -n "${PROFILE}" ] && [ -f "${PROFILE}" ]; then
+    echo "${PROFILE}"
+    return
+  fi
+
+  local DETECTED_PROFILE
+  DETECTED_PROFILE=''
+
+  if [ "${SHELL#*bash}" != "$SHELL" ]; then
+    if [ -f "$HOME/.bashrc" ]; then
+      DETECTED_PROFILE="$HOME/.bashrc"
+    elif [ -f "$HOME/.bash_profile" ]; then
+      DETECTED_PROFILE="$HOME/.bash_profile"
+    fi
+  elif [ "${SHELL#*zsh}" != "$SHELL" ]; then
+    if [ -f "$HOME/.zshrc" ]; then
+      DETECTED_PROFILE="$HOME/.zshrc"
+    elif [ -f "$HOME/.zprofile" ]; then
+      DETECTED_PROFILE="$HOME/.zprofile"
+    fi
+  fi
+
+  if [ -z "$DETECTED_PROFILE" ]; then
+    for EACH_PROFILE in ".PROFILE_PATH" ".bashrc" ".bash_profile" ".zprofile" ".zshrc"
+    do
+      if DETECTED_PROFILE="$(try_profile "${HOME}/${EACH_PROFILE}")"; then
+        break
+      fi
+    done
+  fi
+
+  if [ -n "$DETECTED_PROFILE" ]; then
+    echo "$DETECTED_PROFILE"
+  fi
+}
 
 echo -n "Checking NodeJS installation ... "
 if ! command -v node &> /dev/null
 then
-    echo "Node runtime hasn't been detected, installing via nvm"
+    echo "Node runtime hasn't been detected, installing via nvm."
     if command -v curl &> /dev/null
     then
         curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh | bash
     elif command -v wget &> /dev/null
     then
         wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh | bash
     else
-        echo "Neither curl nor wget command is present, unable to proceed"
+        echo "Neither curl nor wget command is present, unable to proceed."
         exit 1
     fi
 else
     MINIMAL="v16.10.0"
-    DETECTED=`node -v`
+    DETECTED=$(node -v)
     echo $DETECTED
-    RESULT=`node -e "console.log(\"$DETECTED\".replace('v', '').localeCompare(\"$MINIMAL\".replace('v', ''), undefined, { numeric: true }))"`
+    RESULT=$(node -e "console.log(\"$DETECTED\".replace('v', '').localeCompare(\"$MINIMAL\".replace('v', ''), undefined, { numeric: true }))")
     if [ $RESULT -lt 0 ]; then
-        echo "Minimal supported version is" $MINIMAL
+        echo "Minimal supported version is $MINIMAL"
         exit 1
     fi
 fi
 
-echo "Installing the command ..."
-npm install --loglevel=error -g @ibm-generative-ai/cli
+HAS_NVM_INSTALLED=$(command -v nvm);
+PROFILE_PATH=$(detect_profile)
+
+if [ -n "$HAS_NVM_INSTALLED" ]; then
+  echo "Installing @ibm-generative-ai/cli ..."
+  npm install --loglevel error -g @ibm-generative-ai/cli
+else
+  if [ -z "$PROFILE_PATH" ]; then
+    echo "Profile cannot be detected."
+    exit 1
+  fi
```

Comment:
This is not an installation blocker necessarily, just guard the necessary script lines, do not exit.

User will get `"Failure, make sure your PATH is set up correctly"` message later on.

Reply:
Not quite so. The user will receive a permission denied error during the global installation if not run with `sudo`, which is not desired.

Reply:
I though the very purpose of this PR was for the script to succeed even without `sudo`, hence the install dir change (to one for which the user has permissions).